colossalai/zero/sharded_param/sharded_param.py

```diff
@@ -40,3 +40,28 @@ def sharded_data_tensor(self):
     @property
     def param_is_sharded(self):
         return self._sharded_data_tensor.is_sharded
+
+    def get_memory_usage(self) -> Tuple[int, int]:
+        """
+        get the memory usage of the param, including data and grad
+        Returns:
+            Tuple[int, int]: cuda mem usage in Byte, cpu memory usage in Byte
+        """
+        cuda_mem_use, cpu_mem_use = 0, 0
+
+        def _update_mem_use(t: Optional[torch.Tensor]):
+            if t is None:
+                return
+            assert isinstance(t, torch.Tensor)
+            nonlocal cuda_mem_use
+            nonlocal cpu_mem_use
+            if t.device.type == 'cpu':
+                cpu_mem_use += t.numel() * t.element_size()
```

Comment:
Seems it would return the number of Byte, do you consider formatting it to KB/MB/GB?

Reply:
Yeah, we can convert the unit at the time it is used. Currently, we can just make the API as simple as possible.

Reply:
ok